Trust management (#46)
APTITUDE-Consortium/wp2-trust-specifications: docs/topics/trust-management-process.md

﻿

> This section describes the high level trust management process within the APTITUDE Trust Framework. It defines the various entities and their roles within the ecosystem ([Ecosystem Partecipants](#ecosystem-participants)), the relationship between their Attributes, the Trust Artifacts that assert these Attributes ([Entity Attribute Schema](#entity-attribute-schema)), introduces the State Machine that governs the lifecycle of WRPs and Wallet Providers within the ecosystem ([Abstract State Machine](#abstract-state-machine)), and describes the operational procedures triggered by changes in these Attributes ([Entity Lifecycle Operations](#entity-lifecycle-operations)).

Author: ```suggestion
This section describes the high level trust management process within the APTITUDE Trust Framework. 
The main target is the lifecycle of Trust Artifacts related to WRPs and Wallet Providers (Register entries and related WRPAC/WRPRC, LOTE and TLs, Qseals) that assert identity and authorizations on Attestations and Attributes, highlighting their dependencies with 
1. other entities and their roles within the ecosystem ([Ecosystem Partecipants](#ecosystem-participants)), 
2. Attestations and Attributes in the Catalogue of Schemes, 
This chapter introduces the State Machine that governs the lifecycle of WRPs and Wallet Providers within the ecosystem ([Abstract State Machine](#abstract-state-machine)), and describes the operational procedures triggered by dependencies between entities and catalogue of schemes.```

Reviewer: merged locally

> Each WRP and Wallet Provider partaking the Ecosystem is represented by specific *Attributes* (Identity Information, Technical Information, and Policy and Authorization Information) and possesses a *State*. This State is an abstraction at the governance level that captures the entity's current operational status and trustworthiness within the ecosystem. The state of an Entity is determined by the information contained in the Trust Artifacts associated with it, which are in turn issued and managed by other entities within the ecosystem (Trusted Lists and Trust Artifacts Providers).

Author: I think we need to find a synonim of Attribute because in this context could be misleading and misunderstood with attribute of attestation.

Reviewer: switched to Property

> Each WRP and Wallet Provider partaking the Ecosystem is represented by specific *Attributes* (Identity Information, Technical Information, and Policy and Authorization Information) and possesses a *State*. This State is an abstraction at the governance level that captures the entity's current operational status and trustworthiness within the ecosystem. The state of an Entity is determined by the information contained in the Trust Artifacts associated with it, which are in turn issued and managed by other entities within the ecosystem (Trusted Lists and Trust Artifacts Providers).

Author: ```suggestion
Each WRP and Wallet Provider partaking the Ecosystem is represented by specific *Properties* (Identity Information, Technical Information, and Policy and Authorization Information). The operational status, referred as *State* is reflected by the state of related Trust Artifacts.
```

Reviewer: swithched to *Properties*

> When a WRP's, Wallet Provider's, or Trust Artifacts Provider's Attributes changes (e.g., key rotation), it triggers an operational procedure that instructs the Trust Artifact Provider or Trusted Lists Providers to update and re-sign the corresponding Trust Artifact or Trusted Lists, without necessarily altering the underlying Abstract State of the WRP. These operational procedures are defined in [Entity Lifecycle Operations](#entity-lifecycle-operations).

Author: ```suggestion
When a WRP's, Wallet Provider's, or Trust Artifacts Properties change (e.g., key rotation), the Trust Artifact Providers or Trusted Lists Providers have to update, revoke and issue or re-sign the corresponding Trust Artifacts or Trusted Lists, without necessarily altering the underlying Abstract State of the WRP. These operational procedures are defined in [Entity Lifecycle Operations](#entity-lifecycle-operations).
```

Reviewer: merged locally

    sup e1@=="Supervise"==> init_ent
    sup e2@=="Supervise"==> mid_ent
    sup e3@=="Supervise"==> end_ent
    %%sup e8@=="Supervise"==> end_ent_2
    ms --"Provides/Publishes"--> sup_art
    sup_art e4@--"Affects"--> init_ent
    sup_art e5@--"Affects"--> mid_ent
    sup_art e6@--"Affects"--> end_ent
    %%sup_art e7@--"Affects"--> end_ent_2
    mid_ent -."Are included".-> init_ent_art
    end_ent -."Are included".-> init_ent_art
    mid_ent_art --"to"----> end_ent
    init_ent --"Publishes"--> init_ent_art
    %%init_ent --"Impacts"--> mid_ent
    mid_ent --"Issue"--> mid_ent_art
    %%mid_ent --"Impacts"--> end_ent
    end_ent --"Issue"--> end_ent_art --"to"--> wi["Wallet Units"]
    wi["Wallet Units"] --"Present Attestations"--> end_ent
    
    
    e1@{ animate: true }
    e2@{ animation: fast }
    e3@{ animation: fast }
    %%e8@{ animation: fast }
    %%Artifacts inclusion arrows
    %%mid_ent e4@-. Referenced in.-> init_ent_art
    %%end_ent e5@-. Referenced in.-> init_ent_art
    %%end_ent e6@-. Referenced in.-> mid_ent_art
    e4@{ animation: fast }
    e5@{ animation: fast }
    e6@{ animation: fast }
    %%e7@{ animation: fast }
```

#### Entity Attribute Schema

> Trust Artifacts Providers and End-Entities are characterized by three main classes of *Attributes*:

Author: ```suggestion
Trust Artifacts Providers and End-Entities are characterized by three main classes of *Properties*:
```

Reviewer: swhitched locally

- **Identity Information**: This includes the organization's name, contact information, and organizational policies.
- **Technical Configuration**: This includes the cryptographic materials (signature/seal keys, authentication keys) and technical endpoints necessary for ecosystem interactions.
- **Policy and Authorization Information**: This includes the entity's entitlements, attestation provision capabilities, attestation request capabilities, intermediary use permissions, intended use cases, and compliance with certification schemas.

##### Attribute Schema and associated Trust Artifacts

In the table below are found the relationship between the aforementioned Attributes and the Trust Artifacts in which they are contained for both WRP and Wallet Providers.

| Entity Type | Attribute Class | Entity Attributes | Trust Artifacts |
| :--- | :--- | :--- | :--- |
| WRP | Identity Information | Organization name | WRPAC, WRPRC, Register |

> | WRP | Identity Information | Contact information | WRPAC, Register |

Author: there won't be an onboarding for a member state registrar. probably it should be included in tier 1
paragraph pattern: affected entities and then related updated artifacts.

wrp: registrar does onboarding and updates register
wp: acquires certification by CAB + NAB, that notifies lote manager

Reviewer: I agree that probably a Registrar does no onboarding. But since it is a role designated at MS level, and also this action by the registrar should already be factored in the onboarding specifications.

Regarding the WP certification, i've added it in the table.

    tech --"used by"--> acca
    acca --"issues/manages"--> ac

    reg --"used by"--> ca
    tech --"used by"--> ca
    ca --"issues/manages"--> csig

    reg -."used by".-> rcca
    rcca --"issues/manages"--> rc

    id --"contained in"--> reg
    authz --"reflected in"--> tl
    authz --"contained in"--> reg
    abst --"used by<br>(when PID/Pub-EAA Provider)"--> tlp
    tlp --"issues/manages"--> tl
    

    wp["Wallet Provider"]

    subgraph abst_wp["WP Attributes"]
        direction LR
        id_wp(["Identity Information<br>(organization name, contact information, organization policies)"])
        tech_wp(["Technical Configuration<br>(signature/seal key,<br>endpoints)"])
        authz_wp(["Policy and Authorization<br>(wallet attestation provvision, compliance to certification schema)"])
    end

    subgraph arti_wp["WP Artifacts"]
        direction LR
        csig_wp{{"Signature/Seal Certificate"}}
    end
    ca_wp["Signature/Seal<br>Certificate Authority or QTSP"]

    wp --"characterized by"--> abst_wp
    authz_wp --"reflected in"--> tl
    tech_wp --"used by"--> ca_wp
    ca_wp --"issues/manages"--> csig_wp

    abst_wp --"used by"--> tlp
```

> In the table below we map the relationship between the aforementioned Attributes and the Trust Artifacts in which they are contained for Trust Artifacts Providers.

Author: ```suggestion
In the table below we map the relationship between the aforementioned Properties and the Trust Artifacts in which they are contained for Trust Artifacts Providers.
```

| Entity Type | Attribute Class | Entity Attributes | Trust Artifacts |
| :--- | :--- | :--- | :--- |
| Registrar, Provider of WRPAC/WRPRC | Identity Information | Organization name | LoTE |
| Registrar, Provider of WRPAC/WRPRC | Identity Information | Contact information | LoTE |

> | Registrar, Provider of WRPAC/WRPRC | Identity Information | Organizational Policy | LoTE |

Author: The policy should impact only the EDP no?

Reviewer: Added EDP

﻿

> This section describes the high level trust management process within the APTITUDE Trust Framework. Its scope is describing the lifecycle of Entities and Trust Artifacts related to WRP and Wallet Providers within the ecosystem, and the relationships between them. Trust Artifacts are the transport mechanism of the Properties (specific Entity attributes categorized in Identity Information, Technical Information, and Policy and Authorization Information) which characterize the entities within the ecosystem, and they are issued and managed by other entities within the ecosystem (Trusted Lists and Trust Artifacts Providers).

Reviewer 2: This is maybe a stupid point, because everybody should know it if is reading this type of document but from the formal point of view: shall we introduce the acronym WRP?

Reviewer: Hopefully in the final specifications, there will be a glossary pop-up with the definition of WRP. So I've been using acronyms loosely throughout.

In particular, this section is structured as follows:

> - in ([Ecosystem Partecipants](#ecosystem-participants)) are described the various entities within the ecosystem, their roles and relationships.

Reviewer 2: "Partecipants" please correct typo

- in ([Abstract State Machine](#abstract-state-machine)) are described the lifecycle *State* (an abstraction at the governance level that captures the current operational status and trustworthiness within the ecosystem) of WRPs, Wallet Providers, Trust Artifacts, Trusted Lists, their definitions, and the effects that these states have on the entities' operational capabilities and trustworthiness.
- in ([Entity Lifecycle Operations](#entity-lifecycle-operations)) are described the operational procedures triggered by changes in the Properties of WRP and Wallet Providers, and the resulting effects on their lifecycle states and trustworthiness within the ecosystem.

When a WRP's, Wallet Provider's, or Trust Artifacts Properties change (e.g., key rotation), the Trust Artifact Providers or Trusted Lists Providers have to update, revoke and issue or re-sign the corresponding Trust Artifacts or Trusted Lists, without necessarily altering the underlying Abstract State of the WRP. These operational procedures are defined in [Entity Lifecycle Operations](#entity-lifecycle-operations).

> #### Ecosystem Participants

Reviewer 2: Why do you start the ideation with 4#?, shouldn't we start with less "#"'s?

Reviewer: The indentation is consistent with the final specs of which these sections are at `###`

- *Trusted Lists Providers* (LoTE/LOTL Scheme Operators, EUMS TL Scheme Operators) *publish* and *manage* *Trusted Lists* (LoTE, LOTL, EUMS TL) that contain Trust Anchors and Properties of other Entities. Inclusion in these artifacts is, by itself, also a statement about the role and authorization of the included entities within the ecosystem.
- *Trust Artifacts Providers* (MS Registrars, QTSPs, Providers of WRPAC/WRPRC) *publish* and *manage* *Trust Artifacts* (MS Registers, WRPACs, WRPRCs, Signature/Seal Certificates) which transports Properties (Identity Information, Technical configurations, and Authorization Information) of End-Entities.
- *End-Entities* (Attestation Providers, PID Providers, Relying Parties, Wallet Provider) rely on Trusted Lists and Trust Artifacts for assessing their trustworthiness to other participants within the ecosystems. In addition, they *issue*, *receive* or *manage* User Attestations or Wallet Attestations to and from Wallet Units during issuance, presentation and WP-specific management flows respectively.

> The diagram below highlights the relationships between the aforementioned entities and artifacts, and the dependences between them in terms of supervision, publication, and the effects that changes in the artifacts have on the entities. The arrows indicate the direction of supervision, publication, and effect propagation.

Reviewer 2: To make clear to the reader the diagram below we can add something like:
... the aforementioned entities (squares in the diagram) and artifacts (diamonds in the diagram)

Or adding a legend in the diagram will do it as well

Reviewer: Thanks, added a description.

    %%mid_ent e4@-. Referenced in.-> init_ent_art
    %%end_ent e5@-. Referenced in.-> init_ent_art
    %%end_ent e6@-. Referenced in.-> mid_ent_art
    e4@{ animation: fast }
    e5@{ animation: fast }
    e6@{ animation: fast }
    %%e7@{ animation: fast }
```

#### Entity Properties Schema

Trust Artifacts Providers and End-Entities are characterized by three main classes of *Properties*:

- **Identity Information**: This includes the organization's name, contact information, and organizational policies.
- **Technical Configuration**: This includes the cryptographic materials (signature/seal keys, authentication keys) and technical endpoints necessary for ecosystem interactions.
- **Policy and Authorization Information**: This includes the entity's entitlements, attestation provision capabilities, attestation request capabilities, intermediary use permissions, intended use cases, EDPs, and compliance with certification schemas.

##### Properties Schema and associated Trust Artifacts

In the table below are found the relationship between the aforementioned Properties and the Trust Artifacts in which they are contained for both WRP and Wallet Providers.

> | Entity Type | Properties Class | Entity Properties | Trust Artifacts |

Reviewer 2: I a not sure to understand this table, maybe if you want to detail relationshipFor instance you started with WRP description but then PID Provider, then WRP again, and so on.

Shouldn't be better to group the rows by entity type?

On the other hand, we already described this info in WRPAC, WRPRC, etc, so I was wondering of we can some how restructure this table to link with this "pages"

Reviewer: Thanks for the feedback, I will restructure as you suggest the table.

| :--- | :--- | :--- | :--- |
| WRP | Identity Information | Organization name | WRPAC, WRPRC, Register |
| WRP | Identity Information | Contact information | WRPAC, Register |
| WRP | Identity Information | Organizational Policy | WRPAC, WRPRC, Register |
| PID Provider or Attestation Provider | Technical Configuration | Signature/Seal key | Signature/Seal Certificate |
| WRP | Technical Configuration | Authentication key | WRPAC |

> | PID Provider or Attestation Provider | Authorization Information | RP Permissions | EDP |

Reviewer 3: maybe separate the RP from the PID/EAA providers? 
something like: 
| RP / RP Intermediary | Authorization Information | Attestation request capabilities, intended use, legal basis / purpose, requested attribute sets, intermediary permissions | Register, WRPRC, EDP where applicable |
| PID Provider / EAA Provider / Pub-EAA Provider | Authorization Information | Attestation provision capabilities, supported attestation types, issuing authority / authentic source relationship, supervision information | Register, LoTE, Signature/Seal Certificate, WRPRC where applicable |

Reviewer: Table restructured per entity type. This way it will be more readable

| WP, PID and Pub-EAA Provider | Authorization Information | Service status | LoTE |
| WP, PID and Pub-EAA Provider | Authorization Information | Compliance to certification schema | LoTE (implicit via inclusion) |
| WP, PID and Pub-EAA Provider | Technical Configuration | Signature/Seal trust anchors | LoTE |

!!! note

    The inclusion of Wallet Providers and PID/Pub-EAA Providers in the LoTE is an implicit assertion of their role and authorization within the ecosystem. In particular, their inclusion is a result of the succesful completion of the registration and notification procedures as defined in CIR 2025/848 (registration of WRP) and CIR 2024/2980 (notifications of WRP and WP).

The diagram below highlights the dependences between WRP's and Wallet Provider's Properties, the Trusted Lists in which these Properties are contained and the Entities that use the Properties' information to issue additional Trust Artifacts. For both WRPs and WPs, the inclusion in the Trusted Lists attests the implicit, ongoing, compliance to the polices set up by the EC and the MS during onboarding for the respective roles.

> ```mermaid

Reviewer 2: Maybe it would help the inclusion (or description) of a legend, describing the different object and colours of the diagram, don' you think?

Reviewer: I could not find a native way to make mermaid flowchart support diagrams, so i added details in the text

flowchart LR
    wrp["WRP"]

    subgraph abst["WRP Properties"]
        direction LR
        id(["Identity Information<br>(organization name, contact information, organization policies)"])
        tech(["Technical Configuration<br>(signature/seal key,<br>AuthN key, endpoints)"])

>         authz(["Policy and Authorization<br>(entitlements, attestation provvision, attestation request, intermediary use, intended use, compliance to certification schema)"])

Reviewer 2: Please correct "provvision" typo

```mermaid
flowchart LR
    mid_ent["Trust Artifact Provider<br>(Registrar, Provider of WRPAC/WRPRC, QTSP)"]

    subgraph abst["Entity Properties"]
        direction LR
        id(["Identity Information<br>(organization name, contact information, organization policies)"])
        tech(["Technical Configuration<br>(signature/seal key,<br> endpoints)"])
        authz(["Policy and Authorization<br>(compliance to certification schema)"])
    end

    subgraph arti["Entity Artifacts"]
        direction LR
        csig{{"Signature/Seal Certificate"}}
    end

    ms["Member State, European Commission"]

    tl{{"LoTE or EUMS TL"}}

    ca["Signature/Seal<br>Certificate Authority"]
    tlp["LoTE/EUMS TL Scheme Operator"]

    mid_ent --"characterized by"--> abst

    id --"used by"--> ca
    tech --"used by"--> ca
    ca --"issues/manages"--> csig

    abst --"used by"--> tlp
    tlp --"publishes/maintains"--> tl

    ms --"defines"--> authz
    authz --"is reflected in"--> tl
```

> ##### Abstract State Machine

Reviewer 2: It the indention right here?, shouldn't it be 4# not 5?

Reviewer: Corrected

This section describes the lifecycle *State* of WRPs and Wallet Providers and of Trust Artifacts and Trusted Lists.

###### Entity Lifecycle State Machine

> State Machines are described only for WRPs and Wallet Providers since they are the only entities that have a direct, active role in the ecosystem operations and protocols. Wallet Providers and WRPs participating in the APTITUDE Trust Framework are classified into one of the following mutually exclusive lifecycle States at any given time. The State dictates the entity's authorization level, operational capabilities, and how other participants SHALL interact with its cryptographic artifacts.

Reviewer 2: Here you state: "State Machines are described **only** for WRPs and Wallet Providers ....", but hen the next section is "Trust Artifacts and Trusted Lists Lifecycle State Machine". Maybe it is contradictory?

> State Machines are described only for WRPs and Wallet Providers since they are the only entities that have a direct, active role in the ecosystem operations and protocols. Wallet Providers and WRPs participating in the APTITUDE Trust Framework are classified into one of the following mutually exclusive lifecycle States at any given time. The State dictates the entity's authorization level, operational capabilities, and how other participants SHALL interact with its cryptographic artifacts.

Reviewer 3: State machines are described for operational ecosystem entities, including WRPs, Wallet Providers, and Trust Artifact Providers, where their operational status is externally visible through trust artifacts or trusted-list entries.

because later on we have the OPERATIONAL definition includes “Wallet Provider or Trust Artifacts Provider”.

Reviewer: Suggestion incorporated, thanks

    - A WRP is in `REMOVED` state if the WRPAC is `revoked`.
    - A Wallet Provider or Trust Artifacts Provider is in `REMOVED` state when it is not listed in the latest version of the relevant LoTE or EUMS TL.
    - **Forward Operations**: Ecosystem participants SHALL reject new interactions or transactions initiated by a `REMOVED` entity, and all cryptographic keys, active attestations, and operational capabilities associated with the entity SHALL be immediately revoked.
    - **Historical Operations**: Ecosystem participants MAY continue to validate historical data, signatures, and attestations generated prior to the Removal timestamp, subject to local risk policies. These historical data are found in the corresponding Trusted List's `ServiceHistory` component.

Below the state diagram of the various actors.

```mermaid
stateDiagram-v2
    %% Define states
    [*] --> Unreg
    
    state "UNREGISTERED" as Unreg
    state "REGISTERED" as Reg
    state "OPERATIONAL" as Op
    state "REMOVED" as Withdrawn
    
    %% Define transitions
    Unreg --> Reg : Onboarding & Identity Verification<br>(WRPAC Issued & Register Inclusion)
    

>     Reg --> Op : Authorization Granted<br>(Signature/Seal Issued & LoTE ServiceState `granted`)

Reviewer 3: at other points we have ServiceStatus instead? decide on one?

| `REVOKED` | Indicates that a Trust Artifact has been revoked and SHALL NOT be trusted for any operational use within the ecosystem. | WRPAC, WRPRC, Signature/Seal Certificates. | OCSP response with `revoked` status or presence in CRL for WRPACs and Signature/Seal certificates; SLT with status set to `0x01` for WRPRCs. |
| `UPDATED` | Indicates that a Trusted List has been updated with a new version, and the previous version is no longer valid for operational use within the ecosystem. | LoTE, LOTL, EUMS TL, Registers, EDP, Status Lists. | Publication of a new version of the Trusted List; resolution of historical status via the `ServiceHistory` component for LoTE, LOTL, and EUMS TL. |
| `HISTORICAL` | Indicates that a Trusted List is a historical record and SHALL NOT be used for any operational use within the ecosystem, except for validating historical operations and trustworthiness via the pivoting mechanism. | LoTE, LOTL, EUMS TL. | Resolution of historical status via the `ServiceHistory` component; validation of trustworthiness via the pivoting mechanism. |

> | `UPDATED` | Indicates that a Register, EDP, or Status List has been updated with new information, and the previous version is no longer valid for operational use within the ecosystem. | Registers, EDP, Status Lists. | Publication of an updated version of the artifact. |

Reviewer 2: I understand we need to describe two different states 'UPDATED', but to see two "rows" with the same name/state is very weird, maybe we should do 3 different tables to avoid this?

Reviewer: restructured the table and description. Also changed `UPDATED` to `CURRENT` as it feels more term-wise

#### Entity Lifecycle Operations

Below are detailed the operational procedures, which affects the Properties of Trust Artifact Provider and End-Entities and the resulting effects on their State.

##### Onboarding Process

> The onboarding process governs the transition of a End-Entity (WRP or Wallet Provider) from the `UNREGISTERED` state to the `OPERATIONAL` state. During the onboarding process, it is up to the Entities running the process and necessary checks to take the onboardee's Properties and (depending on its role) issue the corresponding Trust Artifacts. Further details are found in [Onboarding Process](#onboarding-process.md).

Reviewer 2: Is this link right? it is pointing itself

Reviewer: It should be right (at least sintax-wise). Unfortunately it points to another md file and the viewer cannot render it properly...

> The onboarding process governs the transition of a End-Entity (WRP or Wallet Provider) from the `UNREGISTERED` state to the `OPERATIONAL` state. During the onboarding process, it is up to the Entities running the process and necessary checks to take the onboardee's Properties and (depending on its role) issue the corresponding Trust Artifacts. Further details are found in [Onboarding Process](#onboarding-process.md).

Reviewer 2: You are stating: "from the `UNREGISTERED` state to the `OPERATIONAL` state." No mention to the `RESGISTERED` state, is that intentional?

##### Active Operations and Maintenance

> While in the `OPERATIONAL` state, entities MAY require *organizational updates* to their registered Properties including, entities profile, cryptographic materials, or operational parameters. Below we describe these updates and their operational effects on the trust artifacts and the entity's state.

Reviewer 2: Is this sentence finished, shouldn't it follow something like this: While ... so/then/but ...

###### Operational Effects of Organizational Updates

When there are organizational updates, the Trust Framework infrastructure SHALL propagate these changes to the relevant trust artifacts. The specific operational effects depend on the entity's role, and the artifacts it utilizes.

**WRP Updates**: For WRP updating their Identity Information, Technical Configurations, and/or Policies and Authorizations, the update event SHALL trigger the following procedures:

> - **Identity Information and Policies and Authorizations Updates**:

Reviewer 2: too many "and"?

    - **Registry Update**: the WRP SHALL update its information within the Register via the authenticated API.
    - **WRPAC Revocation**: The Provider of WRPAC SHALL revoke the entity's current WRPAC. Revocation SHALL be executed by appending the certificate's serial number to the active CRL or by returning a `revoked` status in the OCSP response.

>     - **WRPAC Re-issuance**:The Provider of WRPAC SHALL issue a new WRPAC with the updated data.

Reviewer 2: a "real" event?, only if there is an specific attribute changing the State should be considered (and then it will be specific in other points), right?

    - **WRPRC Revocation**: The Provider of WRPRC SHALL revoke the entity's current WRPRC if present. Revocation SHALL be executed by setting the status value of the WRPRC within the corresponding Status List token to `0x01`.

>     - **WRPRC Re-issuance**: The Provider of WRPRC SHALL issue a new WRPRC with the updated data if required.

Reviewer 2: As this is not necessarily going to change the State, can we consider it as an event?

Reviewer: Removed the event name

    - **Endpoints Update** [ONLY for Registrars or Wallet Providers]: When updating endpoint, the Registrar or Wallet Provider SHALL notify the LoTE Scheme Operator which will publish a new version of the LoTE using the pivoting mechanism with the updated endpoint.

##### Removal Process

The Removal process defines the rapid-response workflows and administrative procedures executed to transition an entity from the `OPERATIONAL` state to the `REMOVED` state. This transition MAY be initiated voluntarily by the entity or forcefully enacted by the Supervisory Body.

###### Triggers for Removal

Removal events are categorized based on their initiation source:

- Voluntary Exit: Organizations MAY choose to exit the federation for standard business or operational reasons. Permitted reasons include:
    - Business Changes: Organizational restructuring, mergers, acquisitions, or complete service discontinuation.
- Supervisory Body Removal (use the MS, EC, CAB, NAB, DPA bodies): The Supervisory Body MAY initiate a forced Removal due to severe compliance failures, fatal security breaches, or other critical ecosystem threats.

###### Operational Effects of Removal

> When an entity is transitioned to the `REMOVED` state, the APTITUDE Trust Framework infrastructure SHALL immediately execute a series of cryptographic and registry updates to halt the entity's operations while preserving historical evidence.  The specific operational effects depend on the entity's role, and the artifacts it utilizes.

Reviewer 2: Sorry what do you mean with "When an entity is transitioned to the `REMOVED` state, the APTITUDE Trust Framework infrastructure SHALL immediately execute ..."? 
I thought this is an analysis of the ARF not about the APTITUDE  Trust Framework, if so, we should mention it at the beginning, don't you think?

Reviewer: Removed the APtitude TRust framework reference

**Trust Artifact Provider Withdrawal or Removal**: For Wallet Providers, Registrars, Providers of WRPAC/WRPRC updating their Identity Information and/or Technical Configurations, the removal event SHALL trigger the following procedures:

- **LoTE Update**: the Trust Artifact Provider or Supervisory body SHALL notify the LoTE Scheme Operator which will publish a new version of the LoTE using the pivoting mechanism where the entity's `TrustedEntityServices.ServiceInformation.ServiceStatus` component SHALL be set to the URI `http://uri.aptitude.org/TrstSvc/TrustedList/Svcstatus/withdrawn`. To maintain non-repudiation for past transactions, the superseded parameters SHALL be retained as historical records within the `TrustedEntityServices.ServiceHistory`.

!!! note

    When executing the revocation on the CRL, the `ReasonFlag` element SHALL accurately reflect the nature of the Removal:

>     - If the Removal is a temporary suspension pending investigation, the `ReasonFlag` SHALL be set to `(6)`: `certificateHold`.

Reviewer 3: I think in the state machine no suspended state is added?  only operational and removed?

Reviewer: Removed the suspension part as agreed today
